streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -161,16 +157,17 @@ private static boolean isRecoverable(final KafkaException uncaughtException) {
      * @throws IllegalStateException if EOS is disabled
      * @throws TaskMigratedException
      */
-    void commitTransaction(final Map<TopicPartition, OffsetAndMetadata> offsets) throws ProducerFencedException {
+    void commitTransaction(final Map<TopicPartition, OffsetAndMetadata> offsets,
+                           final ConsumerGroupMetadata consumerGroupMetadata) throws ProducerFencedException {
         if (!eosEnabled) {
             throw new IllegalStateException(formatException("EOS is disabled"));
         }
         maybeBeginTransaction();
         try {
-            producer.sendOffsetsToTransaction(offsets, applicationId);
+            producer.sendOffsetsToTransaction(offsets, consumerGroupMetadata);
             producer.commitTransaction();
             transactionInFlight = false;
-        } catch (final ProducerFencedException error) {
+        } catch (final ProducerFencedException | CommitFailedException error) {
```

Comment:
Not for this PR: `FencedInstanceIdException` should not be handled as a task-migration exception since it indicates some other instance with the same instance.id has started, and hence this instance should really be closed as a whole.

Just to clarify your PR did the right thing to not capture it, I just set it as a reminder for myself since I found that some other places we capture `FencedInstanceIdException` incorrectly.